packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -156,36 +156,51 @@ class Dart2JSTarget extends Target {
     final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
     final String specPath = globals.fs.path.join(globals.artifacts.getArtifactPath(Artifact.flutterWebSdk), 'libraries.json');
     final String packageFile = PackageMap.globalPackagesPath;
+    final File outputKernel = environment.buildDir.childFile('app.dill');
     final File outputFile = environment.buildDir.childFile('main.dart.js');
 
-    final ProcessResult result = await globals.processManager.run(<String>[
+    // Run the dart2js compilation in two stages, so that icon tree shaking can
+    // parse the kernel file for web builds.
```

Comment:
Upstream tests of const finder are all against `--aot --tfa`.  It would probably be good to add some upstream tests for this case as well since the kernel format is different.  We don't have any real tests of the correctness of const_finder in this repo, and assume the upstream tests in the engine would catch anything incorrect.